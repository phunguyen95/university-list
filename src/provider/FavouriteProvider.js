import React, { useState, useEffect } from "react";
import _isEmpty from "lodash/isEmpty";
export const favouriteUniList = React.createContext();

const FavouriteProvider = (props) => {
  const [favouriteList, setFavouriteList] = useState([]);
  return (
    <favouriteUniList.Provider value={{ favouriteList, setFavouriteList }}>
      {props.children}
    </favouriteUniList.Provider>
  );
};
export default FavouriteProvider;
