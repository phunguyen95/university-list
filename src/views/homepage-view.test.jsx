import { render, screen } from "@testing-library/react";
import HomepageView from "./homepage-view";
import FavouriteProvider from "../provider/FavouriteProvider";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

test("renders error view without crashing", () => {
  const history = createMemoryHistory();
    const favouriteList = [];
    const setFavouriteList = jest.fn();
  const { asFragment } = render(
    <Router history={history}>
      <FavouriteProvider value={{ favouriteList, setFavouriteList }}>
        <HomepageView />
      </FavouriteProvider>
    </Router>
  );

  expect(asFragment()).toMatchSnapshot();
});
