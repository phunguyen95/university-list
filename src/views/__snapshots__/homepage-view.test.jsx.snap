// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error view without crashing 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="title"
    >
      University List
    </h1>
    <div
      class="field"
    >
      <p
        class="control is-expanded has-icons-right"
      >
        <input
          class="input searchInput"
          placeholder="Search..."
          type="search"
          value=""
        />
        <span
          class="icon is-small is-right"
        >
          <i
            class="fas fa-search"
          />
        </span>
      </p>
    </div>
    <section
      class="section"
    >
      <h4>
        Total 0 seach results
      </h4>
      <label
        class="label sortBy"
      >
        Sort By:
      </label>
      <div
        class="select is-normal"
      >
        <select>
          <option
            value="country"
          >
            Country
          </option>
          <option
            selected=""
            value="name"
          >
            Name
          </option>
        </select>
      </div>
      <div
        class="level-item"
      >
        <small>
          Page 0 of 1 
        </small>
      </div>
      <div
        class="container"
      >
        <div
          class="b-table"
        >
          <div
            class="table-wrapper has-mobile-cards"
          >
            <table
              class="table is-fullwidth is-striped is-hoverable is-fullwidth"
            >
              <thead>
                <tr>
                  <th>
                    No.
                  </th>
                  <th>
                    Name
                  </th>
                  <th>
                    Country
                  </th>
                  <th>
                    Website
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  class="is-empty"
                >
                  <td
                    colspan="7"
                  >
                    <section
                      class="section"
                    >
                      <div
                        class="content has-text-grey has-text-centered"
                      >
                        <p>
                          <span
                            class="icon is-large"
                          >
                            <i
                              class="mdi mdi-emoticon-sad mdi-48px"
                            />
                          </span>
                        </p>
                        <p>
                          Nothing's thereâ€¦
                        </p>
                      </div>
                    </section>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="notification"
          >
            <div
              class="level"
            >
              <div
                class="level-left"
              >
                <div
                  class="level-item"
                >
                  <div
                    class="buttons has-addons"
                  />
                </div>
                <div
                  class="level-right"
                >
                  <button
                    class="pagination-previous"
                    disabled=""
                  >
                    Previous
                  </button>
                  <button
                    class="pagination-next"
                    disabled=""
                  >
                    Next page
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
